Scripts/Level_Components/IDE.gd

```diff
@@ -76,7 +109,36 @@ func run_code():
 				yield(get_tree().create_timer(GameStats.run_speed, false), "timeout") 
 			elif block.BLOCK_TYPE == "CALL":
 				var call_name = block.name.trim_prefix("Call_").rstrip("0123456789").trim_suffix("_")
+				if is_loop(call_name):
+					scopes[call_name].reset_loopCount()
+					looping = scopes[call_name].check_conditions()
 				yield(enter_scope(scopes[call_name]), "completed")
+				
+		#Increment loopCount for For Loops, refresh code array, & continue Loop or stop it
+		if looping:
```

Comment:
If (looping):
---Yield
---If For loop, then Increment
---looping = check conditions //Kicks into effect on the next loop
Else (not looping):
---code = context.pop_back()
---If (code.back is a loop) //Check if we just returned back into a loop
------loopNode = code.back
------looping = true

Reply:
Just realized, the current Pseudocode is only good for For loops rn.